src/ArtView/ArtView.tsx

```diff
@@ -19,45 +33,53 @@ interface ImageDetails {
 }
 
 interface Props {
-  imageId: string;
-  artPieces: {
-    id: number;
-    image_src: string;
-    description: string;
-    title: string;
-    date_start: number;
-    image_id: string;
-    artist_title: string;
-    classification_titles: Array<string>;
-  }[]
+  category: string;
 }
 
 class ArtView extends Component<Props, State> {
   constructor(props: Props) {
     super(props)
     this.state = {
       userLikesPhoto: undefined,
-      singlePiece: undefined
+      singlePiece: undefined,
+      isLoading: true,
+      artPieces: [],
+      randomImageId: ''
     }
   }
 
-  findPiece = (value: boolean) => {
-    const foundPiece = this.props.artPieces.find(piece => piece.image_id === this.props.imageId)
-    this.setState({ userLikesPhoto: value, singlePiece: foundPiece })
+  componentDidMount = () => {
```

Comment:
I really like how you had the loading icon show up based on the isLoading property, which changes when state is set after the promise is resolved successfully.